packages-exp/analytics-exp/src/initialize-ids.ts

```diff
@@ -0,0 +1,137 @@
+/**
+ * @license
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {
+  DynamicConfig,
+  Gtag,
+  MinimalDynamicConfig
+} from '@firebase/analytics-types-exp';
+import { GtagCommand, GA_FID_KEY, ORIGIN_KEY } from './constants';
+import { FirebaseInstallations } from '@firebase/installations-types-exp';
+import { getId } from '@firebase/installations-exp';
+import { fetchDynamicConfigWithRetry } from './get-config';
+import { logger } from './logger';
+import { FirebaseApp } from '@firebase/app-types-exp';
+import {
+  isIndexedDBAvailable,
+  validateIndexedDBOpenable
+} from '@firebase/util';
+import { ERROR_FACTORY, AnalyticsError } from './errors';
+
+async function validateIndexedDB(): Promise<boolean> {
+  if (!isIndexedDBAvailable()) {
+    logger.warn(
+      ERROR_FACTORY.create(AnalyticsError.INDEXEDDB_UNAVAILABLE, {
+        errorInfo: 'IndexedDB is not available in this environment.'
+      }).message
+    );
+    return false;
+  } else {
+    try {
+      await validateIndexedDBOpenable();
+    } catch (e) {
+      logger.warn(
+        ERROR_FACTORY.create(AnalyticsError.INDEXEDDB_UNAVAILABLE, {
+          errorInfo: e
+        }).message
+      );
+      return false;
+    }
+  }
+  return true;
+}
+
+/**
+ * Initialize the analytics instance in gtag.js by calling config command with fid.
+ *
+ * NOTE: We combine analytics initialization and setting fid together because we want fid to be
+ * part of the `page_view` event that's sent during the initialization
+ * @param app Firebase app
+ * @param gtagCore The gtag function that's not wrapped.
+ * @param dynamicConfigPromisesList Array of all dynamic config promises.
+ * @param measurementIdToAppId Maps measurementID to appID.
+ * @param installations FirebaseInstallations instance.
+ *
+ * @returns Measurement ID.
+ */
+export async function initializeIds(
```

Comment:
Can we give the function a more descriptive name, e.g. describing what specific things are being initialized, or maybe give it a more high level name, e.g. initializeAnalytics()?

Reply:
`initializeAnalytics` seems good.